packages/PriceLockup/__tests__/PriceLockup.spec.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import Text from '@tds/core-text'
+import HairlineDivider from '@tds/core-hairline-divider'
+import PriceLockup from '../PriceLockup'
+
+const defaultProps = {
+  signDirection: 'left',
+  price: '25',
+  rateText: '/month',
+  bottomText: 'bottom text',
+}
+
+describe('PriceLockup', () => {
+  const doShallow = (props = {}) => shallow(<PriceLockup {...defaultProps} {...props} />)
+  const doMount = (props = {}) => mount(<PriceLockup {...defaultProps} {...props} />)
+
+  it('renders with default props', () => {
+    const priceLockup = doShallow()
+    expect(priceLockup).toMatchSnapshot()
+  })
+
+  it('does not allow custom CSS', () => {
+    const priceLockup = doShallow({
+      className: 'my-custom-class',
+      style: { color: 'hotpink' },
+    })
+    expect(priceLockup).not.toHaveProp('className', 'my-custom-class')
+    expect(priceLockup).not.toHaveProp('style')
+  })
+
+  describe('size prop is small', () => {
+    const priceLockup = doShallow({ size: 'small' })
+    it('will make price h2 sized', () => {
+      expect(priceLockup.html()).toContain('priceValueSignsmall')
+    })
+    it('will make rate text medium sized', () => {
+      priceLockup.setProps({ rateText: '/month' })
+      expect(priceLockup).toContainReact(<Text size="medium">/month</Text>)
+    })
+    it('will make top text small sized', () => {
+      priceLockup.setProps({ topText: 'top text' })
+      expect(priceLockup).toContainReact(<Text size="small">top text</Text>)
+    })
+  })
+
+  describe('size prop is medium', () => {
+    const priceLockup = doShallow({ size: 'medium' })
+    it('will make price h2 sized', () => {
+      expect(priceLockup.html()).toContain('priceValueSignmedium')
+    })
+    it('will make rate text medium sized', () => {
+      priceLockup.setProps({ rateText: '/month' })
+      expect(priceLockup).toContainReact(<Text size="medium">/month</Text>)
+    })
+    it('will make top text small sized', () => {
+      priceLockup.setProps({ topText: 'top text' })
+      expect(priceLockup).toContainReact(<Text size="small">top text</Text>)
+    })
+    it('will make bottom text medium sized', () => {
+      priceLockup.setProps({ bottomText: 'bottom text' })
+      expect(priceLockup).toContainReact(<Text size="medium">bottom text</Text>)
+    })
+  })
+
+  describe('size prop is large', () => {
+    const priceLockup = doShallow({ size: 'large' })
+    it('will make DollarSign h1 sized', () => {
+      expect(priceLockup.html()).toContain('dollarSignH1Style')
+    })
+    it('will make price h2 sized', () => {
+      expect(priceLockup.html()).toContain('priceValueSignlarge')
```

Comment:
Try following my comment above for this test as well. Please refactor all tests above and below this line accordingly.

Reply:
done